### PR TITLE
BDR-621: Initial pass of lazy iterator chunking interface

### PR DESCRIPTION
**Pull request by HaydenGaia**
## Description
* Changes `ABISMapper.apply_mapping()` interface to return an `Iterator[rdflib.Graph]` instead of just `rdflib.Graph`
* Added argument `chunk_size: Optional[int]` to `ABISMapper.apply_mapping()` to specify chunk size
* Implementors are expected to provide a lazy iterator interface for mappers
* Added `utils.chunking.should_chunk()` utility function to determine when to yield a chunk

### CODE REVIEW

File: abis_mapping/templates/dwc_mvp/mapping.py [hunk 1]
```diff
@@ -105,6 +106,10 @@ def apply_mapping(
             onerror="raise",  # Raise errors, it should already be valid here
         )
 
+        # Infer Statistics and Count Number of Rows
+        resource.infer(stats=True)
```
Comment (kehh) [inline abis_mapping/templates/dwc_mvp/mapping.py:110]: How safe is this `infer` method? Is it part of frictionless?

File: abis_mapping/utils/chunking.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+
+# Third-Party
+import frictionless
+
+# Typing
+from typing import Optional
+
+
+def should_chunk(
```
Comment (kehh) [inline abis_mapping/utils/chunking.py:11]: Probably need a test for this but I'm sure you've got that on your radar.

**Review by kehh — COMMENTED** [review]
Some questions and comments

File: abis_mapping/templates/dwc_mvp/mapping.py (hunk 1 above)
Comment (HaydenGaia) [reply abis_mapping/templates/dwc_mvp/mapping.py:110]: @kehh Yes, from `frictionless`: https://framework.frictionlessdata.io/docs/guides/framework/resource-guide/#stats

File: abis_mapping/utils/chunking.py (hunk 2 above)
Comment (HaydenGaia) [reply abis_mapping/utils/chunking.py:11]: @kehh Good point, missed them for some reason

Comment (HaydenGaia) [reply abis_mapping/utils/chunking.py:11]: Unit tests added in 18147e7

**Pull request merged**